File: src/main/java/com/kamikazejam/datastore/base/cache/StoreLoader.java
package com.kamikazejam.datastore.base.cache;

import com.kamikazejam.datastore.base.Store;

import java.util.Optional;

/**
 * This class is responsible for loading a {@link Store} when requested from a Cache.
 */
@SuppressWarnings("rawtypes")
public interface StoreLoader<X extends Store> {

    Optional<X> fetch(boolean saveToLocalCache);

}

File: src/main/kotlin/com/kamikazejam/datastore/base/cache/StoreLoader.kt
package com.kamikazejam.datastore.base.cache

import com.kamikazejam.datastore.base.Cache
import com.kamikazejam.datastore.base.Store

/**
 * This class is responsible for loading a [Store] when requested from a [Cache].
 */
interface StoreLoader<X : Store<*, *>?> {
    fun fetch(saveToLocalCache: Boolean): X?
}
